include/dlaf/multiplication/general/impl.h

```diff
@@ -19,13 +20,49 @@
 #include <dlaf/communication/communicator_grid.h>
 #include <dlaf/matrix/distribution.h>
 #include <dlaf/matrix/index.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_ref.h>
 #include <dlaf/matrix/panel.h>
 #include <dlaf/multiplication/general/api.h>
 #include <dlaf/sender/when_all_lift.h>
+#include <dlaf/util_matrix.h>
 
 namespace dlaf::multiplication {
 namespace internal {
 
+template <Backend B, Device D, class T>
+void General<B, D, T>::callNN(const T alpha, MatrixRef<const T, D>& mat_a, MatrixRef<const T, D>& mat_b,
+                              const T beta, MatrixRef<T, D>& mat_c) {
+  namespace ex = pika::execution::experimental;
+
+  DLAF_ASSERT_HEAVY(matrix::multipliable(mat_a, mat_b, mat_c, blas::Op::NoTrans, blas::Op::NoTrans),
+                    mat_a, mat_b, mat_c);
+
+  if (mat_a.nrTiles().cols() == 0) {
+    // Note: if beta == 1, we optimize by not even scheduling anything
+    if (beta != T(1)) {
+      for (SizeType j = 0; j < mat_c.nrTiles().cols(); ++j)
+        for (SizeType i = 0; i < mat_c.nrTiles().rows(); ++i)
+          ex::start_detached(dlaf::internal::whenAllLift(beta, mat_c.readwrite(GlobalTileIndex(i, j))) |
```

Comment:
Probably better to switch to:
- local instead of global
- and new snake_case functions

```suggestion
      for (SizeType j = 0; j < mat_c.distribution().local_nr_tiles().cols(); ++j)
        for (SizeType i = 0; i < mat_c.distribution().local_nr_tiles().rows(); ++i)
          ex::start_detached(dlaf::internal::whenAllLift(beta, mat_c.readwrite(LocalTileIndex(i, j))) |
```